src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -407,7 +407,10 @@ UCS_PROFILE_FUNC_ALWAYS(ucs_status_t, uct_ib_mlx5_devx_reg_atomic_key,
                         (md, memh), uct_ib_mlx5_md_t *md,
                         uct_ib_mlx5_devx_mem_t *memh)
 {
-    uct_ib_mr_type_t mr_type  = uct_ib_md_get_atomic_mr_type(&md->super);
+    /* dm only supports default mr */
+    uct_ib_mr_type_t mr_type  = memh->dm != NULL ?
+                                         UCT_IB_MR_DEFAULT :
+                                         uct_ib_md_get_atomic_mr_type(&md->super);
```

Comment:
shall we pass memh->dm or memh to uct_ib_md_get_atomic_mr_type?

Reply:
yes, but it will have to be in a new function because `uct_ib_md_get_atomic_mr_type` is in /ib level and here we want to pass devx memh